narwhals/series.py

```diff
@@ -71,7 +71,14 @@ def __arrow_c_stream__(self, requested_schema: object | None = None) -> object:
         native_series = self._compliant_series._native_series
         if hasattr(native_series, "__arrow_c_stream__"):
             return native_series.__arrow_c_stream__(requested_schema=requested_schema)
-        pa = get_pyarrow()
+        try:
+            import pyarrow as pa
+        except ModuleNotFoundError as exc:  # pragma: no cover
+            msg = f"PyArrow>=14.0.0 is required for `__arrow_c_stream__` for object of type {type(native_series)}"
+            raise ModuleNotFoundError(msg) from exc
+        if parse_version(pa.__version__) < (14, 0):  # pragma: no cover
+            msg = f"PyArrow>=14.0.0 is required for `__arrow_c_stream__` for object of type {type(native_series)}"
+            raise ModuleNotFoundError(msg)
         ca = pa.chunked_array([self.to_arrow()])
```

Comment:
I think this might require pyarrow 15

Reply:
in pandas the requirement is PyArrow 14+ (I also just ran the tests with pyarrow 13 and 14 - the former fails, the latter passes)

Reply:
ah sorry, that's for DataFrame

Reply:
It was added to pa.chunked_array in a later release, yes. I think it was here: https://github.com/apache/arrow/pull/40818